superset/message_threads/commands/create.py

```diff
@@ -0,0 +1,81 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import logging
+from typing import Optional, Union
+
+from superset.commands.base import BaseCommand, CreateMixin
+from superset.daos.exceptions import DAOCreateFailedError
+from superset.daos.thread import MessageThreadDAO
+from superset.message_threads.models import MessageThread
+
+logger = logging.getLogger(__name__)
+
+
+class CreateThreadCommand(BaseCommand):
+    def __init__(
+        self,
+        workspace_id,
+        chart_id: Optional[int],
+        dashboard_id: Optional[int],
+        id: Optional[int],
+    ):
+        self.workspace_id = workspace_id
+        self._chart_id = chart_id
+        self._dashboard_id = dashboard_id
+        self._id = id
+
+    def run(self) -> Union[MessageThread, None]:
+        # self.validate()
+        try:
+            # this is a singleton command. It will return
+            # the existing thread if it exists
+            thread = MessageThreadDAO.get(
+                workspace_id=self.workspace_id,
+                chart_id=self._chart_id,
+                dashboard_id=self._dashboard_id,
+                id=self._id,
+            )
+            if thread:
+                return thread
+            elif self._dashboard_id and self._chart_id:
+                thread = MessageThreadDAO.create(
+                    workspace_id=self.workspace_id,
+                    chart_id=self._chart_id,
+                    dashboard_id=self._dashboard_id,
+                )
+                return thread
```

Comment:
we may need to return a flag if this is new so that the client side knows to generate a new thread on the cord side.